app/src/main/java/com/babylon/wallet/android/presentation/account/composable/MetadataView.kt

```diff
@@ -0,0 +1,333 @@
+                    contentDescription = null,
+                    tint = iconColor
+                )
+            }
+
+            MetadataType.String -> ExpandableText(
+                modifier = modifier,
+                text = metadata.value,
+                style = style.copy(
+                    color = color,
+                    textAlign = if (isRenderedInNewLine) TextAlign.Start else TextAlign.End,
+                ),
+                toggleStyle = style.copy(color = iconColor)
+            )
+
+            MetadataType.Instant -> {
+                val displayable = remember(metadata.value) {
+                    val epochSeconds = metadata.value.toLongOrNull() ?: return@remember metadata.value
+                    val dateTime = Instant.ofEpochSecond(epochSeconds)
+                        .atZone(ZoneId.systemDefault())
+                        .toLocalDateTime()
+
+                    dateTime.format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM))
+                }
+                Text(
+                    modifier = modifier,
+                    text = displayable,
+                    style = style,
+                    color = color,
+                    textAlign = if (isRenderedInNewLine) TextAlign.Start else TextAlign.End,
+                    maxLines = 2
+                )
+            }
+
+            MetadataType.Address -> ActionableAddressView(
+                modifier = modifier,
+                address = remember(metadata.value) {
+                    Address.init(metadata.value)
+                },
+                textStyle = style,
+                textColor = color,
+                iconColor = iconColor
+            )
+
+            MetadataType.NonFungibleGlobalId -> ActionableAddressView(
+                modifier = modifier,
+                globalId = remember(metadata.value) {
+                    NonFungibleGlobalId.init(metadata.value)
+                }.copy(),
+                showOnlyLocalId = false,
+                textStyle = style,
+                textColor = color,
+                iconColor = iconColor
+            )
+
+            MetadataType.NonFungibleLocalId -> Text(
+                modifier = modifier,
+                text = remember(metadata.value) {
+                    NonFungibleLocalId.init(metadata.value).formatted(AddressFormat.DEFAULT)
+                },
+                style = RadixTheme.typography.body1HighImportance,
+                color = RadixTheme.colors.gray1,
+                textAlign = if (isRenderedInNewLine) TextAlign.Start else TextAlign.End,
+                maxLines = 2
+            )
+
+            MetadataType.Decimal -> Text(
+                modifier = modifier,
+                // If value is unable to transform to big decimal we just display raw value
+                text = metadata.value.toDecimal192OrNull()?.formatted() ?: metadata.value,
```

Comment:
It would be great to use the default dispatcher for Sargon function calls, especially here because this view serves as a list item so we potentially might have many Sargon function calls on the UI thread, affecting the performance.

Reply:
You are right let me change that